main.cpp

```diff
@@ -119,288 +154,357 @@ struct JumpStarter
-    4)material
-    5)number of clicks before failure
-3 things it can do:
-    1)reset the script
-    2)produce a click sound
-    3)trigger a led
- */
+
+int Memory::writeScript( int scriptSize )
+{
+    if( scriptSize < size/2 )
+    return size - scriptSize;
+    else return size;
+}
+
+int Memory::writeData( int location , int dataSize )
+{
+    if( location < size - dataSize )
+    return location + dataSize;
+    else return 0;
+}
+
+int Memory::writeStatus( int statusCode )
+{
+    if( statusCode > 100 )
+    return 100;
+    else return statusCode;
+}
+///////////////////////////////////////////////////////////////////////
 struct ResetButton
 {
-    //1)size
     float height = 0.05f;
-    //2)location on the board
     char location = 't';
-    //3)color
     char color = 'r';
-    //4)material
     char material = 'p';
-    //5)number of clicks before failure
     int clicksBeforeFailure = 10000;
 
-    //1)reset the script
     int triggerResetSginal( float clickTime = 0 );
-    //2)produce a click sound
     void clickSound( float clickTime = 0 );
-    //3)trigger a led
     void triggerLed( float clickTime = 0 );
 };
-/*
-10)microcontroller
-5 properties:
-    1)USB interface
-    2)converters
-    3)serial port
-    4)memory
-    5)reset button
-3 things it can do:
-    1)store instructions received from computer trough USB
-    2)get a data character from serial interface
-    3)read analog value from AD converter 
- */
+
+int ResetButton::triggerResetSginal( float clickTime )
+{
+    int dummy;
+
+    if( clickTime > 0 )
+    {
+        dummy = 1;
+        return 1;
+    }
+    return 0;
+}
+
+void ResetButton::clickSound( float clickTime )
+{
+    int x;
+    int dummy = 0;
+
+    if( clickTime > 0 )
+    {
+        for( x = 0; x < 100; x++ )
+        { dummy++;} 
+    }
+
+}
+void ResetButton::triggerLed( float clickTime )
+{
+    int dummy = 0;
+
+    if( clickTime > 0 )
+    {
+        dummy = 1;
+    }
+}
+
+///////////////////////////////////////////////////////////////////////
 struct Microcontroller
 {
-    //1)USB interface
     UsbInterface usb0;
-    //2) converter
     Converter ad;
     Converter da;
-    //3)
     SerialPort serialPort;
-    //4)memory
     Memory memory;
-    //5)reset button
     ResetButton resetButton;
 
-    struct Script      //this was not requested but can be used later
+    struct Script      
     {
         int scriptLength; 
     };
 
-    //1)store instructions received from computer trough USB
     int storeScript( Script script );
-    //2)get a data character from serial interface
     char getCharacter( SerialPort serialPort, int channel = 1 ); 
-    //3)read analog value from AD converter 
-    char readValue( Converter ad, int channel = 0 ); 
+    int readValue( Converter ad, int channel = 0 ); 
 };
+
+int Microcontroller::storeScript( Script script )
+{
+    int dummy = 0;
+    int x = 0;
+
+    for( x = 0; x < script.scriptLength + 1; x++ )
+    {
+        dummy = x;
+    }
```

Comment:
- x is already 0;  don't do `for( x = 0;`.  just do `for( ; x < ...`
- why not just do `dummy = script.scriptLength + 1` since that's what it ends up being when the loop ends